tests/core/test_exp_manager.py

```diff
@@ -0,0 +1,232 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import shutil
+from pathlib import Path
+
+import pytest
+import pytorch_lightning as pl
+from omegaconf.errors import OmegaConfBaseException
+
+from nemo.utils import logging
+from nemo.utils.exp_manager import (
+    CheckpointMisconfigurationError,
+    LoggerMisconfigurationError,
+    NotFoundError,
+    exp_manager,
+)
+
+
+@pytest.fixture
+def cleanup_local_folder():
+    # Asserts in fixture are not recommended, but I'd rather stop users from deleting expensive training runs
+    assert not Path("./lightning_logs").exists()
+    assert not Path("./NeMo_experiments").exists()
+
+    yield
+
+    if Path("./lightning_logs").exists():
+        shutil.rmtree('./lightning_logs')
+    if Path("./NeMo_experiments").exists():
+        shutil.rmtree('./NeMo_experiments')
+
+
+class TestExpManager:
+    @pytest.mark.unit
+    def test_omegaconf(self):
+        """Ensure omegaconf raises an error when an unexcepted argument is passed"""
+        with pytest.raises(OmegaConfBaseException):
+            exp_manager(None, {"unused": 1})
+
+    @pytest.mark.unit
+    def test_trainer_loggers(self, cleanup_local_folder, tmp_path):
+        """ Test that a trainer with logger errors out with a number of arguments. Test that it works with
+        create_tensorboard_logger set to False
+        """
+        test_trainer = pl.Trainer()  # Should create logger and modelcheckpoint
+
+        with pytest.raises(LoggerMisconfigurationError):  # Fails because exp_manager defaults to trainer
+            exp_manager(test_trainer, {"root_dir": str(tmp_path)})
```

Comment:
@blisc I have a slight problem with `root_dir` as the name of directory where the experiments will be stored - for me `root_dir` should indicate the root directory of the project, e.g. `/home/tkornuta/workspace/nemo`.

How about changing it to something that will reflect more role of this parameters (e.g. `experiments_dir`)?

Reply:
Ok, let's change it to exp_dir

Reply:
Changed